interfaces/cython/cantera/composite.py

```diff
@@ -691,7 +707,11 @@ def append(self, state=None, **kwargs):
                     "{} is not a valid combination of properties for setting "
                     "the thermodynamic state".format(tuple(kwargs))
                 ) from None
-            setattr(self._phase, attr, [kwargs[a] for a in attr])
+            if not normalize:
+                setattr(self._phase, attr[-1] + "_no_norm", kwargs[attr[-1]])
+                setattr(self._phase, attr[:-1], [kwargs[a] for a in attr[:-1]])
+            else:
+                setattr(self._phase, attr, [kwargs[a] for a in attr])
```

Comment:
Likewise here about the reversed logic and the `attr` checking. Another potential simplification here, specifically, might be:

```python
if attr[-1] == "X":
    setattr(self._phase, "no norm mole fractions", kwargs.pop("X"))
elif attr[-1] == "Y":
    setattr(...)
setattr(self._phase, attr[:2], list(kwargs.values()))
```
and a similar change for the normalized case. I'm not sure if that'll pull the right values though. You might even be able to get away with not casting the `kwargs.values()` to a `list`, depending on how that ends up being used in the setters, but I'm not sure about that.